src/shared/discover-image.c

```diff
@@ -1255,7 +1256,8 @@ int image_read_metadata(Image *i, const ImagePolicy *image_policy) {
 
                 r = dissected_image_acquire_metadata(m,
                                                      DISSECT_IMAGE_VALIDATE_OS |
-                                                     DISSECT_IMAGE_VALIDATE_OS_EXT);
+                                                     DISSECT_IMAGE_VALIDATE_OS_EXT,
+                                                     &image_class);
```

Comment:
rather than carrying this around, add it to the image class

Reply:
unsure what this means.

Reply:
add it as a field in the DissectImage struct